Batch8CucumberSeleniumAPI/src/test/java/utils/ui/Interact.java

```diff
@@ -0,0 +1,88 @@
+package utils.ui;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.By;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+/**
+ * 
+ * @author akashtyagi
+ * Library created to interact with the elements
+ */
+public abstract class Interact {
+
+	private static final Logger logger = LogManager.getLogger(Interact.class);
+	private WebDriver driver;
+	
+	public WebDriver getDriver() {
+		 return driver;
+	}
+	
+	public void setDriver(WebDriver d) {
+		this.driver=d;
+	}
+	
+	public WebElement clickElement(By by) {
+		WebDriverWait wait = new WebDriverWait(driver, 60);
+		WebElement element = wait.until(ExpectedConditions.elementToBeClickable(by));
+		element.click();
+		logger.info("Element is clicked. Element Description: " + by.toString());
+		return element;
+	}
+	
+	public WebElement setElement(By by, String text) {
+		WebDriverWait wait = new WebDriverWait(driver, 60);
+		WebElement element = wait.until(ExpectedConditions.elementToBeClickable(by));
+		element.sendKeys(text);
+		logger.info("Element is Set with text as: " + text + ". Element Description: " + by.toString());
+		return element;
```

Comment:
what is the purpose of return element.

Reply:
I am returning this object of time WebElement in case it is required outside. However, i am not using it. But just given an option if any one wishes to access it. Other wise it's of no specific use.